### ISSUE TITLE
vscode: unable to verify token

### ISSUE THREAD

**Issue by ptxmac** [sourcegraph/sourcegraph#34338]
- **Sourcegraph version:** sourcegraph.com
- **Platform information:** vscode extension v2.2.1

#### Steps to reproduce:

1. Try to log in to sourcegraph using a access token
2. It fails with "Unable to verify your access token"

#### Expected behavior:

The vscode extension should send an authenticated request to the server

#### Actual behavior:

I used `nc` to open a tpc server and looked at the request the extension is sending:

```
$ nc -l  8899
POST /.api/graphql?CurrentAuthState HTTP/1.1
0: Authorization
0: token 123
accept-encoding: gzip,deflate
user-agent: node-fetch/1.0 (+https://github.com/bitinn/node-fetch)
connection: close
accept: */*
content-length: 731
Host: localhost:8899

{"query":"\n    query CurrentAuthState {\n        currentUser {\n            __typename\n            id\n            databaseID\n            username\n            avatarURL\n            email\n            displayName\n            siteAdmin\n            tags\n            url\n            settingsURL\n            organizations {\n                nodes {\n                    id\n                    name\n                    displayName\n                    url\n                    settingsURL\n                }\n            }\n            session {\n                canSignOut\n            }\n            viewerCanAdminister\n            tags\n            tosAccepted\n            searchable\n        }\n    }\n","variables":{}}^C⏎
```

Note in the top that the Authorization header is not send correctly - it looks like the input is some sort of array (?) being marshalled to text incorrectly.

The correct way to send HTTP authorization headers is  `Authorization: token 123`

My guess is that there are a bug somewhere in the configuration of the HTTP client used by the vscode extension

**Comment by abeatrix** [comment]
Hi @ptxmac , since I'm unable to reproduce the issue, can you tell us more about where you are seeing the error to make sure we are on the same page?

I am able to log in after entering my Sourcegraph Access Token in the sidebar.

However, if I enter an incorrect Access Token, the error message you describe would show up:
![image](https://user-images.githubusercontent.com/68532117/165362848-b0c87a13-09fd-44f5-8152-e502e80f5e00.png)

If this is the case, can you try logging in again using a new Access Token created in your Sourcegraph's User Setting and let us know if that helps?

Is your Access Token created on www.sourcegraph.com, or is it associated with a private instance?

If it is associated with a private instance, you will also to configure the instance url by:
1. Click on `Need to connect to a private instance?`
![image](https://user-images.githubusercontent.com/68532117/165363634-7793958b-aebf-4cd8-ac2f-c83a79927e27.png)

2. Enter the relevant `Access Token` and `Sourcegraph Instance URL` here:
![image](https://user-images.githubusercontent.com/68532117/165363740-034a6fdf-61c9-4115-b8b9-41ab6ca6ada6.png)

I hope this helps!

**Comment by sourcegraph-bot-2** [comment]
Heads up @muratsu @jjinnii @ryankscott - the "team/integrations" label was applied to this issue.

**Comment by ptxmac** [comment]
The issue is not with token validation, but the way the token is sent to the server - from the output of `nc` it's clear that the header is not added correctly.

From some debugging, I found that changing how the HTTP request is made solves the problem, see #34389

But - the original code _should_ work - So I'm not sure why my VSCode is behaving differently.
One suspect would be another extension that is overriding the `fetch` function. 

Unfortunately, I have no idea how to debug that

**Comment by abeatrix** [comment]
@ptxmac Just want to let you know that the fix has been merged and it should be included in the next patch release (2.2.3). Thanks again for bringing this to our attention and for the suggestion!
